### ISSUE TITLE
Pulumi CLI should warn about expired AWS credentials or docker daemon not running

### ISSUE THREAD

**Issue by Hansesto** [pulumi/pulumi#6458]
Hello,

*TL;DR*: Abort and notify/warn user when running update, refresh or other command in case the AWS credentials are not set, or if they have expired. Also in case docker daemon needs to be running, but is not.

-----

I usually like pulumi, and everything has gone fine thus far. However, now that I was going to perform an update on a stack (on AWS) adding a couple of resources, the `pulumi refresh`  just got stuck, and just blinked the "refreshing.." texts on the resources. I waited for over an hour, and nothing seemed to happen.

I canceled the refresh and tried to force deploy the resources without checking if they are up-to-date with their actual resource counterparts on AWS (i.e. by not doing a refresh), with `pulumi up`. This got stuck as well. I tried to log information with `pulumi up --skip-preview --logtostderr -v=9 2> pulumi_out.log`, but couldn't find out why the update was stuck.

I tried to look for troubleshooting tips etc. but couldn't find out what was wrong.

After a while of debugging, I remembered that docker daemon wasn't running, which is required for building images (that the stack uses). I thought that this would solve the issues, but pulumi still got stuck performing the mentioned commands. Without any informative error messages, just "updating.." and "refreshing..".

Finally, when I ran another (not related in any way to pulumi or the stack) script that uses AWS credentials (as does pulumi), it notified me that my AWS credentials had expired. Once I refreshed the credentials (and now that the docker daemon was running), I was finally able to use pulumi again.

I know that these are prerequisities for using pulumi (docker not necessarily, if not required by some specific resources). But if these are forgotten, it would be very helpful if pulumi would notify the user if the used AWS credentials have expired, or if docker daemon is not running. Instead of just being stuck and showing the "refreshing.." lines to the user.

**Comment by lukehoban** [comment]
I believe this is due to the issue(s) discussed in https://github.com/pulumi/pulumi-aws/issues/814#issuecomment-597991455.  

We've heard more reports of this recently though, so I'll keep this issue open to investigate again what we can do in the near term to address this.
